test/recipes/80-test_pkcs12.t

```diff
@@ -170,6 +170,59 @@ ok(grep(/Trusted key usage (Oracle)/, @pkcs12info) == 0,
     ok(scalar @match > 0 ? 0 : 1, "test_export_pkcs12_outerr6_empty");
 }
 
+# Test export of PEM file with both cert and key, with password.
+# PBMAC1 protection, default algorithms
+{
+    my $pbmac1_id = "pbmac1_defaults";
+    ok(run(app(["openssl", "pkcs12", "-export", "-pbmac1",
+            "-inkey", srctop_file(@path, "cert-key-cert.pem"),
+            "-in", srctop_file(@path, "cert-key-cert.pem"),
+            "-passout", "pass:1234",
+            "-out", "$pbmac1_id.p12"], stderr => "${pbmac1_id}_err.txt")),
+    "test_export_pkcs12_${pbmac1_id}");
+    open DATA, "${pbmac1_id}_err.txt";
+    my @match = grep /:error:/, <DATA>;
+    close DATA;
+    ok(scalar @match > 0 ? 0 : 1, "test_export_pkcs12_${pbmac1_id}_err.empty");
+}
+
+# Test export of PEM file with both cert and key, with password.
+# PBMAC1 protection, non-default algorithms
+{
+    my $pbmac1_id = "pbmac1_nondefaults";
+    ok(run(app(["openssl", "pkcs12", "-export", "-pbmac1",
+            "-inkey", srctop_file(@path, "cert-key-cert.pem"),
+            "-in", srctop_file(@path, "cert-key-cert.pem"),
+            "-passout", "pass:1234",
+            "-pbmac1_pbkdf2_kdf_md", "sha512", "-macalg", "sha384",
+            "-out", "$pbmac1_id.p12"], stderr => "${pbmac1_id}_err.txt")),
+    "test_export_pkcs12_${pbmac1_id}");
+    open DATA, "${pbmac1_id}_err.txt";
+    my @match = grep /:error:/, <DATA>;
+    close DATA;
+    ok(scalar @match > 0 ? 0 : 1, "test_export_pkcs12_${pbmac1_id}_err.empty");
+}
```

Comment:
These two tests are very similar - I suggest factoring out the common part.

Reply:
Moreover, looks like the produced PEM output including the generated MAC is not checked, but just that no error occurred.
Please add a verification step.

Reply:
Yes, some verification is needed

Reply:
Okay

Reply:
I've factored out the common part and added some basic info tests